editor/plugins/particle_process_material_editor_plugin.h

```diff
@@ -0,0 +1,138 @@
+/**************************************************************************/
+/*  particle_process_material_editor_plugin.h                             */
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef PARTICLE_PROCESS_MATERIAL_EDITOR_PLUGIN_H
+#define PARTICLE_PROCESS_MATERIAL_EDITOR_PLUGIN_H
+
+#include "editor/editor_plugin.h"
+#include "editor/editor_properties.h"
+
+class Button;
+class EditorSpinSlider;
+class Label;
+class ParticleProcessMaterial;
+class Range;
+class VBoxContainer;
+
+class ParticleProcessMaterialMinMaxPropertyEditor : public EditorProperty {
+	GDCLASS(ParticleProcessMaterialMinMaxPropertyEditor, EditorProperty);
+
+	enum {
+		HOVER_NONE,
+		HOVER_LEFT,
+		HOVER_RIGHT,
+		HOVER_MIDDLE,
+	};
+
+	enum {
+		DRAG_NONE,
+		DRAG_LEFT,
+		DRAG_RIGHT,
+		DRAG_MIDDLE,
+		DRAG_SCALE,
+	};
+
+	enum {
+		MODE_RANGE,
+		MODE_PIVOT,
```

Comment:
I think we shouldn't use different terms in the code and the UI. Maybe call it `MODE_VALUE_SPREAD` or `MODE_BASE_VALUE_SPREAD`?

Regarding the other occurences of "pivot" (method and member variable names), I'm not sure how to name them the best since (base)value_spread is a bit too long and doesn't fit everywhere. I know this is a bit contradicting with what I said above, but maybe use "midpoint" in those cases? That could be a bit clearer than pivot.